cylc/flow/xtriggers/wall_clock.py

```diff
@@ -27,3 +29,44 @@ def wall_clock(trigger_time=None):
             Trigger time as seconds since Unix epoch.
     """
     return time() > trigger_time
+
+
+def validate(f_args, f_kwargs, f_signature):
+    """Validate and manipulate args parsed from the workflow config.
+
+    wall_clock()  # infer zero interval
+    wall_clock(PT1H)
+    wall_clock(offset=PT1H)
+
+    The offset must be a valid ISO 8601 interval.
+
+    If f_args used, convert to f_kwargs for clarity.
+
+    """
+
+    n_args = len(f_args)
+    n_kwargs = len(f_kwargs)
+
+    if n_args + n_kwargs > 1:
+        raise WorkflowConfigError(f"Too many args: {f_signature}")
+
+    if n_kwargs:
+        # sole kwarg must be "offset"
+        kw = next(iter(f_kwargs))
+        if kw != "offset":
+            raise WorkflowConfigError(f"Illegal arg '{kw}': {f_signature}")
+
+    elif n_args:
```

Comment:
It's valid because of line 50. i.e. it has to be one or the other because the the config file can only have (e.g.)  `wall_clock(PT1H)` or `wall_clock(offset=PT1H)`.

Reply:
Yes, that makes sense - I was projecting backward from the xrandom test where I started out with a copy of this.